### PR TITLE
Ref and repo data aggregation

### PR DESCRIPTION
**Pull request by benmandrew**
Initial work on data aggregation. Currently working on aggregating ref-level data (derived from the head commit of the ref) and repo-level data (derived from its default branch).

To support history, a second data structure is required - already partially implemented by the `Status_cache` - which needs to provide commit-level granularity. The aggregation of ref and repo data will probably rely on this cache for the head commits.

### CODE REVIEW

File: lib/index.mli [hunk 1]
```diff
@@ -111,3 +111,23 @@ val get_active_refs : Repo_id.t -> ref_info Ref_map.t
...
+  val get_ref_started_at : ref_state -> float
+  val get_ref_ran_for : ref_state -> float
+  val get_repo_status : repo_state -> status
+  val get_repo_started_at : repo_state -> float
+
+  val set_ref_state :
+    repo:Repo_id.t -> ref:string -> status -> float -> float -> unit
+
+  val get_ref_state : repo:Repo_id.t -> ref:string -> ref_state
+  val aggregate_repo : repo:Repo_id.t -> unit
```
Comment (novemberkilo) [inline lib/index.mli:132]: Seeing that this function updates the repo's state in the `Repo_map` what do you think of naming this as `update` or `set` or similar?

**Comment by benmandrew** [comment]
Partial solution to #593; work on the DB schema is still necessary.

**Review by novemberkilo — APPROVED** [review]
Thanks @benmandrew - I've skimmed the PR and things look good. I haven't had time to test it locally or look through it in great detail but I don't think this should hold you up from merging it when you are ready to.  I will revisit it and look through things in more detail in the next day or so.

**Comment by benmandrew** [comment]
This PR has been deployed.
(Clarification due to potentially confusing commit history.)

**Pull request merged**